### PR TITLE
Added example about always test `.send` and pulled out to new section

### PR DESCRIPTION
**Pull request by nemild**
For *always test send*, I added an example and pulled out to a new section

### CODE REVIEW

File: smart-contracts.md [hunk 1]
```diff
@@ -89,19 +89,28 @@ There is a slight difference in Solidity as to what happens with Contract calls
 
 The most important point is the same for both, whether using ExternalContract.doSomething() or address.call(), if ExternalContract is untrusted, assume that malicious code will execute.  Note that if you trust an ExternalContract, you also trust any external contracts it calls, and that those call, are all non-malicious.  If any malicious contract exists in the call chain, the malicious contract can attack you (next section on Reentrant Attacks).
 
-The return value of all raw external calls should be checked to see if it failed or not.
+### Always test if `.send` succeeded
```
Comment (simondlr) [inline smart-contracts.md]: I would keep "external calls" here. It's just not about send() here. All external raw calls.

Comment (ethers) [inline smart-contracts.md]: I think talking about send() here is good.  If this PR is merged, I was going to remove the call() example, which is a separate issue and submit another PR to see what you think.  call() goes against the recommendation mentioned about fallback functions: we should also see call() quite sparingly.

Comment (simondlr) [inline smart-contracts.md]: Hmmm. This part is explicitly about checking if it failed or not, which is required for call stack attacks. So talking about send() vs call.value() can be in another spot?

Maybe, slightly rewrite into: "Check all external calls. Use call.value()() sparingly. send() is safer since it doesn't forward gas".

Thoughts?

**Comment by nemild** [comment]
@simondlr agreed, I tried to make a few changes - does this get close to what you're thinking?

@ethers Thanks for the note. I actually have a branch with a section saying favor `.send` over raw calls. I'll send that in as a pull request and you can review. Additionally, once all the pull requests are incorporated, we should do a full read through.
